### PR TITLE
[WIP] feature/create quote price endpoint

### PR DESCRIPTION
**Pull request by AlexSJ1990**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/web/ProductPricingController.java [hunk 1]
```diff
@@ -26,6 +27,7 @@
 import org.slf4j.MDC;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/web/ProductPricingController.java]: ?

File: src/main/java/com/hedvig/productPricing/service/web/dto/QuotePriceResponseDto.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.hedvig.productPricing.service.web.dto;
+
+data class QuotePriceResponseDto (
+  val price: Double
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/web/dto/QuotePriceResponseDto.kt]: `MonetaryAmount` or `BigDecimal` plz :) double is super-duper-unsafe for money

File: src/main/java/com/hedvig/productPricing/service/web/ProductPricingController.java [hunk 3]
```diff
@@ -281,6 +285,82 @@ private boolean validateProductTypeUnderwritingLimits(WebOnboardingQuoteRequest
     }
   }
 
+  @PostMapping("/insurance/getHomeQuotePrice")
+  public ResponseEntity<HomeQuotePriceResponseDto> priceFromProductPricingForHomeQuote(
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/service/web/ProductPricingController.java]:
```suggestion
  public ResponseEntity<HomeQuotePriceResponseDto> getHomeQuotePrice(
```

File: src/main/java/com/hedvig/productPricing/service/web/dto/ProductType.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.hedvig.productPricing.service.web.dto
+
+enum class ProductType {
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/web/dto/ProductType.kt]: Do we need another enum like this, why can't we use the one we have?

File: src/main/java/com/hedvig/productPricing/service/aggregates/MemberAggregate.java [hunk 5]
```diff
@@ -226,11 +226,16 @@ void signedContract(SingedContractCommand cmd, PdfCreator pdfCreator) throws IOE
...
 
-    LocalDateTime activeFrom = this.currentInsurer == null
+    LocalDateTime activeFrom;
+    if(cmd.getActiveFrom() != null){
+      activeFrom = cmd.getActiveFrom();
+    }else {
+    activeFrom = this.currentInsurer == null
       ? LocalDateTime.ofInstant(Instant.now(), ZoneOffset.UTC)
       : null;
+    }
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/aggregates/MemberAggregate.java]: just some details but could we check the formatting of this pls :)

File: src/main/java/com/hedvig/productPricing/service/web/dto/RapioQuoteRequest.kt [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import java.time.LocalDateTime
+
+data class RapioQuoteRequest(
+  val currentTotalPrice: BigDecimal,
+  val firstName: String,
+  val lastName: String,
+  var birthDate: LocalDate,
+  var isStudent: Boolean,
+  val address: Address,
+  var livingSpace: Float,
```
Comment (palmenhq) [inline src/main/java/com/hedvig/productPricing/service/web/dto/RapioQuoteRequest.kt]: why `var`?

**Pull request merged**